### PR TITLE
react-query: type safety for `QueryKey`

### PR DESCRIPTION
**Pull request by stefanthaler**
cc: @turansky

### CODE REVIEW

File: buildSrc/src/main/kotlin/karakum/query/Type.kt [hunk 1]
```diff
@@ -6,6 +6,8 @@ private val SPECIAL_TYPES = setOf(
     "TOutput | DataUpdateFunction<TInput, TOutput>",
 )
 
+private const val QUERY_KEY="QueryKey"
```
Comment (turansky) [inline buildSrc/src/main/kotlin/karakum/query/Type.kt:9]: Formatting problem

File: buildSrc/src/main/kotlin/karakum/query/Type.kt [hunk 2]
```diff
@@ -105,6 +109,12 @@ class Type(
         if (body.toIntOrNull() != null)
             return "const val $name = $body"
 
+        if (name == "QueryKey")
```
Comment (turansky) [inline buildSrc/src/main/kotlin/karakum/query/Type.kt:112]: Const?

File: buildSrc/src/main/kotlin/karakum/query/Type.kt [hunk 3]
```diff
@@ -105,6 +109,12 @@ class Type(
...
+            return """
+                // $body 
+                external interface QueryKey
```
Comment (turansky) [inline buildSrc/src/main/kotlin/karakum/query/Type.kt:115]: `$name`

**Review by turansky — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/karakum/query/Type.kt (hunk 2 above)
Comment (stefanthaler) [reply buildSrc/src/main/kotlin/karakum/query/Type.kt]: Done.

File: buildSrc/src/main/kotlin/karakum/query/Type.kt (hunk 1 above)
Comment (stefanthaler) [reply buildSrc/src/main/kotlin/karakum/query/Type.kt]: Fixed.

File: buildSrc/src/main/kotlin/karakum/query/Type.kt (hunk 3 above)
Comment (stefanthaler) [reply buildSrc/src/main/kotlin/karakum/query/Type.kt]: I wanted the comment to say the original types, i.e. `// string | readonly unknown[] `

File: buildSrc/src/main/kotlin/karakum/query/Type.kt [hunk 4]
```diff
@@ -109,10 +112,10 @@ class Type(
         if (body.toIntOrNull() != null)
             return "const val $name = $body"
 
-        if (name == "QueryKey")
+        if (name == QUERY_KEY)
             return """
                 // $body 
-                external interface QueryKey
+                external interface $QUERY_KEY
```
Comment (turansky) [inline buildSrc/src/main/kotlin/karakum/query/Type.kt:118]: `$name`?

**Review by turansky — CHANGES_REQUESTED** [review]

**Review by turansky — APPROVED** [review]

**Pull request merged**